### PR TITLE
Reorganize the Directory Structure of the Trace files

### PR DESCRIPTION
**Pull request by ardelato**
## Description

This pull request adds a new formatting fixture for trace file output paths in Playwright tests. It modifies the `testInfo.outputDir` property to create a nested directory structure that makes it easier to locate trace files for specific tests.

The new directory structure for trace files will look like this:
```bash
[subtest-directory]/test-file-name/test-name/device-name/
```

For example:
```bash
product/add_to_cart/back-to-shopping-button-works/desktop-chrome/
```

This nested directory structure will allow us to easily find the trace files for a given test as the original trace files were underneath a single directory where the directory name was a flattened version of the new directory structure.

|Before|After|
|:---:|:---:|
|  image  | image     |

### QA Notes

- [ ] Verify that a trace file is created in the expected folder path for a test file in the playwright directory.
	- Check if a `.spec.ts` file lives in the `playwright` folder so that the `[subtest-directory]` path will be omitted and the trace file will still successfully be outputted
- [ ] Test the proper formatting of directory names when special characters are used in test names.
	- Basically check `'` and other special characters that would be used in a test case name will be properly formatted for the use of a directory name

Connects: #1496

### CODE REVIEW

File: frontend/tests/playwright/fixtures/trace-override.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import path from 'node:path';
+
+export function traceOutputDirTemplate(testInfo: TestInfo) {
+   // The path to where all the trace files will be outputted as defined
+   // in the Playwright config
+   const globalOutputDir = testInfo.project.outputDir;
+
+   // titlePath houses the path to the test file relevant to the test directory
+   // e.g. 'product/add_to_cart.spec.ts' -> 'product/add_to_cart'
+   const testFile = testInfo.titlePath[0].replace(/\.spec\.ts$/, '');
```
Comment (mlahargou) [inline frontend/tests/playwright/fixtures/trace-override.ts:11]: I know we only have `spec.ts` files right now. Do you think that will change? Maybe we add `spec.tsx` at some point?

**Review by mlahargou — APPROVED** [review]
CR 📱 with a minor comment. Fixtures are so powerful!

File: frontend/tests/playwright/fixtures/trace-override.ts (hunk 1 above)
Comment (ardelato) [reply frontend/tests/playwright/fixtures/trace-override.ts:11]: > I know we only have spec.ts files right now. Do you think that will change?

I don't think it will change, there might be a chance someone who adds a `.test.ts` file but that would assume they didn't notice the common usage of `.spec.ts` and we can comment on in CR.

As for the `spec.tsx`, the usage of `tsx` would imply that there are some `react` elements used in the test file which `playwright` is not capable of parsing. This is what caused this issue https://github.com/iFixit/react-commerce/issues/1400

Comment (mlahargou) [reply frontend/tests/playwright/fixtures/trace-override.ts:11]: Sounds good :)

**Comment by deltuh-vee** [comment]
`Clicking + and - Buttons Changes Item Quantity in Cart` gets processed as clicking-and-buttons-changes-item-quantity-in-cart. Would we want to not use "+" and "-" characters?

**Comment by mlahargou** [comment]
Yeah, let's just change the test name or forget about it. I don't think it's a huge deal.

**Comment by deltuh-vee** [comment]
QA 🎬
* Traces are created in the expected folder path
* Directory names are not heavily affected by test case names
